CommonTools/PileupAlgos/plugins/PuppiPhoton.cc

```diff
@@ -186,6 +186,7 @@ void PuppiPhoton::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
 }
 // ------------------------------------------------------------------------------------------
 bool PuppiPhoton::matchPFCandidate(const reco::Candidate *iPF,const reco::Candidate *iPho) { 
+  if(iPF->pdgId() != iPho->pdgId()) return false;
   double lDR = deltaR(iPF->eta(),iPF->phi(),iPho->eta(),iPho->phi());
   for(unsigned int i0 = 0; i0 < pdgIds_.size(); i0++) {
     if(std::abs(iPF->pdgId()) == pdgIds_[i0] && lDR < dRMatch_[i0])  return true;
```

Comment:
the change above apparently at least partially invalidates the logic here, where other pdgIds are checked specifically.
Should the loop over pdgIds_ be removed?

I thought that matching of a photon to an electron were still meaningful, but maybe I don't have a good memory of what is done in PuppiPhoton